django_kafka/retry/__init__.py

```diff
@@ -0,0 +1,60 @@
+from typing import TYPE_CHECKING, Optional, Type
+
+from django.utils import timezone
+
+if TYPE_CHECKING:
+    from django_kafka.topic import Topic
+
+
+class RetrySettings:
+    def __init__(
+        self,
+        max_retries: int,
+        delay: int,
+        backoff: bool = False,
+        include: Optional[list[Type[Exception]]] = None,
+        exclude: Optional[list[Type[Exception]]] = None,
+    ):
+        """
+        :param max_retries: max number of retry attempts
+        :param delay: delay (seconds)
+        :param backoff: exponential backoff
+        :param include: exception types to retry for
+        :param exclude: exception types to exclude from retry
+        """
+        if max_retries <= 0:
+            raise ValueError("max_retries must be greater than zero")
+        if delay <= 0:
+            raise ValueError("delay must be greater than zero")
+        if include is not None and exclude is not None:
+            raise ValueError("cannot specify both include and exclude")
+
+        self.max_retries = max_retries
+        self.delay = delay
+        self.backoff = backoff
+        self.include = include
+        self.exclude = exclude
+
+    def attempts_exceeded(self, attempt):
+        return attempt >= self.max_retries
+
+    def should_retry(self, exc: Exception) -> bool:
+        if self.include is not None:
+            return any(isinstance(exc, e) for e in self.include)
+        if self.exclude is not None:
+            return not any(isinstance(exc, e) for e in self.exclude)
+
+        return True
+
+    def get_retry_timestamp(self, attempt: int) -> str:
+        delay = self.delay * 2 ** (attempt - 1) if self.backoff else self.delay
+        return str(timezone.now().timestamp() + delay)
+
+
+class Retry:
+    def __call__(self, *args, **kwargs):
```

Comment:
I didn't really like that I by wrapping `RetrySettings.__init__`, I lose static type checking when using the `kafka.retry(...)` decorator (since everything is passed masked by `args` and `kwargs`). I looked pretty deep in and couldn't find a "simple" way to forward the type args of `RetrySettings.__init__` to the decorator (`__call__`). If you think you know a way?

Reply:
I'd say just duplicate the types

As an alternative we may just get rid of `RetrySettings` and merge the functionality into the `Retry`

`topic_cls.retry_settings = self`